crates/sui-types/src/intent.rs

```diff
@@ -1,51 +1,107 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use eyre::eyre;
+use fastcrypto::encoding::decode_bytes_hex;
 use serde::{Deserialize, Serialize};
 use serde_repr::Deserialize_repr;
 use serde_repr::Serialize_repr;
+use std::str::FromStr;
 
 #[cfg(test)]
 #[path = "unit_tests/intent_tests.rs"]
 mod intent_tests;
 
+/// The version here is to distinguish between signing different versions of the struct
+/// or enum. Serialized output between two different versions of the same struct/enum
+/// might accidentally (or maliciously on purpose) match.
 #[derive(Serialize_repr, Deserialize_repr, Copy, Clone, PartialEq, Eq, Debug, Hash)]
 #[repr(u8)]
 pub enum IntentVersion {
     V0 = 0,
 }
 
+impl From<u8> for IntentVersion {
+    fn from(version: u8) -> Self {
+        match version {
+            0 => Self::V0,
+            _ => panic!("Invalid IntentVersion"),
```

Comment:
Who can invoke that? is it possible that someone exploits it to panic Sui?

Reply:
good point, changed this to try_from. the caller (cli command) of the panic here does not panic sui, but it doesnt prevent future callers. using try_from is safer.